src/lib/generic-views.tsx

```diff
@@ -35,3 +38,26 @@ export class PreRenderedView implements IView {
     }
 
 }
+
+/**
+ * An SVG `foreignObject` view with a namespace specified by the provided `ForeignObjectElement`.
+ * Note that `foreignObject` may not be supported by all browsers or SVG viewers.
+ */
+@injectable()
+export class ForeignObjectView implements IView {
+    render(model: ForeignObjectElement, context: RenderingContext): VNode {
+        const foreignObjectContents = virtualize(model.code);
+        const node = <g>
+            <foreignObject requiredFeatures='http://www.w3.org/TR/SVG11/feature#Extensibility'
+                height={model.bounds.height} width={model.bounds.width} x={0} y={0}>
+                {foreignObjectContents}
+            </foreignObject>
+            {context.renderChildren(model)}
+        </g>;
+        setAttr(node, 'class', model.type);
+        const subType = getSubType(model);
+        if (subType) setAttr(node, 'class', subType);
```

Comment:
The subtype is added for every view in `CssClassPostprocessor` so it's not be needed here.